### DISCUSSION TITLE
Is there a way to run <script setup> only on the server side?

### DISCUSSION THREAD

**Discussion by stausmar** [nuxt/nuxt#16323] [Questions]
As far as I know, `<script setup>` runs on the server once, and then on the client afterwards.
Is there a possibility to **run this only on the server**?

**Comment by tylerforesthauser** [comment]
AFAIK, you can't only run setup on the server but you can certainly run everything _in_ setup on the server...

```
if (process.server) {
  // your code
}
```

**Reply by stausmar** [reply]
@tylerforesthauser - Reading data from an API is something I would prefer doing on the server, always. Is there any other possiblity instead of fetching an `/server/api` endpoint? E.g.:

```js
<script setup>
const { data } = await useFetch('/server/api/foo')
</script>
```
Or do you see any other way to achieve this?

**Reply by tylerforesthauser** [reply]
This is untested but perhaps you could leverage `useState` for this purpose?

```js
const apiData = useState('api-data')
const apiError = useState('api-error')
if (process.server) {
  try {
    const response = await $fetch('/server/api/foo')
    apiData.value = response.data
  } catch (error) {
    apiError.value = error
  }
}
```
